src/test/java/be/syntra/devshop/DevshopBack/services/ShopOrderServiceTest.java

```diff
@@ -0,0 +1,60 @@
+package be.syntra.devshop.DevshopBack.services;
+
+import be.syntra.devshop.DevshopBack.entities.ShopOrder;
+import be.syntra.devshop.DevshopBack.entities.User;
+import be.syntra.devshop.DevshopBack.services.utilities.CartMapper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import static be.syntra.devshop.DevshopBack.testutilities.CartUtils.createCart;
+import static be.syntra.devshop.DevshopBack.testutilities.UserUtils.createUser;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ShopOrderServiceTest {
+
+    @Mock
+    private UserServiceImpl userService;
+
+    @Mock
+    private CartMapper cartMapper;
+
+    @InjectMocks
+    private CartServiceImpl cartService;
+
+    @BeforeEach
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void saveCartToArchivedCartsTest() {
+        // given
+        ShopOrder dummyShopOrder = createCart();
+        User dummyUser = createUser();
+        String name = "one";
```

Comment:
for increased understanding of this test, I would suggest "email". Or inline createUser().getEmail()

Reply:
used dummyUser.getEmail() everywhere now